sorc/lsm_routines.fd/noah.fd/set_soilveg_snippet.f90

```diff
@@ -1,29 +1,34 @@
 !> @file
 !> @brief Routine to set Noah LSM soil and veg params needed for sflx_snippet
 !> @author Clara Draper
-
 !> Below was extracted from namelist_soilveg.f and set_soilveg.f 
 !! (couldn't get above to compile for doxygen)
+!> @author Yuan Xue: add Noah-MP LSM soil and veg params needed for global_cycle
+!> Noah-MP related parameters were extracted from noahmp_table.f
+!> isot (soil type) = 1: STATSGO must be selected if NoahMP is used
+!> ivet (vegetation type) = 1: IBGP is used by UFS offline Land DA for Noah-MP
+!> as of 07/13/2023
 
 module set_soilveg_snippet_mod
 
  implicit none
 
  private
 
- public set_soilveg
+ public set_soilveg_noah
+ public set_soilveg_noahmp
 
 contains
 
 !> This subroutine initializes soil and vegetation
-!! parameters needed in global_cycle/land_increment.f90 
+!! parameters needed in global_cycle/land_increment.f90 for noah
 !! @param[in] isot Soil type
 !! @param[in] ivet Vegetation type
 !! @param[out] maxsmc Maximum soil moisture for each soil type
 !! @param[out] bb B exponent for each soil type
 !! @param[out] satpsi Saturated matric potential for each soil type
 !! @param[out] iret Return integer
-subroutine set_soilveg(isot,ivet, maxsmc, bb, satpsi, iret) 
+subroutine set_soilveg_noah(isot,ivet, maxsmc, bb, satpsi, iret) 
```

Comment:
This is great, thanks. I'll ask you in person about what has changed (specifically, what errors might be in my experiments so far).

Reply:
Explained.